### PR TITLE
added validation for mind name

### LINKED ISSUE #30: Improve Minds name requirements
## Short description of current behavior
Improve the name requirements for minds

## Steps 🕵️‍♂️ 🕵️‍♀️
- Add name requirements.
- Mind name should start with a letter and contain only letters, numbers or underscore, with a maximum of 32 characters. Spaces are not allowed.

## 📢 Additional rewards 🏆
This issue brings 4️⃣ 0️⃣  points for SWAG and entry into the draw for a 👇 :

3️⃣ ❌ 💻 [Razer Blade 16 Gaming Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0510TEN3-R3U1)
👕 🐻 [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
ℹ️ For more info check out https://mindsdb.com/hacktoberfest 👈

### PR DESCRIPTION
**Pull request by Better-Boy**
Added validation for mind name according to the requirements.

Fixes #30

### CODE REVIEW

File: minds/minds.py
Comment (ea-rus) [inline minds/minds.py:225]: let's remove it and check for `.drop`
let's keep check of name only for cases when this name is going to be stored in database:
- create
- update

File: minds/minds.py
Comment (ea-rus) [inline minds/minds.py:274]: we can move this exception into validate_mind_name, to not duplicate messages

**Review by ea-rus — CHANGES_REQUESTED** [review]

File: minds/minds.py
Comment (Better-Boy) [reply minds/minds.py:29]: I will remove the check. Shouldn't the invalid mind names be changed to valid ones?

File: minds/minds.py
Comment (Better-Boy) [reply minds/minds.py]: Done

File: minds/minds.py
Comment (Better-Boy) [reply minds/minds.py]: Done

File: minds/minds.py
Comment (ea-rus) [reply minds/minds.py]: no, let it be kept as is

**Review by ea-rus — APPROVED** [review]

**Comment by Better-Boy** [comment]
@ea-rus can you merge the PR as it's approved?

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by ZoranPandovski** [comment]
@Better-Boy Can you merge main in this branch so the CLA bot run. Once signed, I will merge all pending PRs

**Comment by Better-Boy** [comment]
> @Better-Boy Can you merge main in this branch so the CLA bot run. Once signed, I will merge all pending PRs

Done, but the bot is waiting for an approval from a maintainer.

**Comment by Better-Boy** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ZoranPandovski** [comment]
recheck

**Pull request merged**
